### PR TITLE
Edit release notes checklist in PR template

### PR DESCRIPTION
**Pull request by colebow**
## Description

Update the PR template to use a GitHub task list in the release notes section, and also add an option for "this is a docs change and no release notes are required," because docs are user-visible but don't require release notes.

## Non-technical explanation

Update PR template.

## Release notes

(x) This is not user-visible and no release notes are required.
( ) This is a docs change and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by colebow** [comment]
Something worth pointing out:

image

We're going to see lots of 1/4 tasks complete on PRs, not sure if that's necessarily an issue, but it's at least a little annoying/suboptimal.

**Review by mosabua — CHANGES_REQUESTED** [review]
We should NOT use `[]` since the interferes with the GitHub actions reporting for build and CLA check and so on. We tried that initially with the template and it did not go well.

**Comment by colebow** [comment]
Are you sure? From the tests that @findepi and I have run, it doesn't seem like it's getting in the way?

**Comment by mosabua** [comment]
> Are you sure? From the tests that @findepi and I have run, it doesn't seem like it's getting in the way?

Hm .. interesting.. maybe they updated how this works. In either case .. since this list is a choice and not a task list I still think the reporting is misleading and we are better off not doing that. Is there any issue with the current setup and e.g. use `(x)` or (✅ ) or even just suggesting usage of ✅ and ❌ ?

**Comment by colebow** [comment]
I think I'm in agreement that task lists are not good, because GitHub integrates and visualizes them in a misleading way.

**Comment by findepi** [comment]
@colebow i suggest checkboxes because

- it allows easy edit in an intuitive way. Placing `x` within misaligned (non-monosapce font rendered) braces is not intuitive
- it is very easy to read, as marked checkbox stands out


the interference with actions checkboxes or other aspects of PR status -- please do not worry about that. I'd let @trinodb/maintainers worry about that. It's their job to understand whether a thing is merge-ready.


**IF** we start to use checkboxes for things related to PR status **THEN** we may want to edit docs-related PR template, as it would **THEN** interfere with how we interpret the PR status / merge-readiness.

**Review by findepi — CHANGES_REQUESTED** [review]
.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 ## Release notes
 
 ( ) This is not user-visible and no release notes are required.
+( ) This is a docs change and no release notes are required.
```
Comment (findepi) [inline .github/pull_request_template.md:16]: Don't add more bullet points. Just reword the first one.

**Review by ebyhr — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -12,7 +12,7 @@
 <!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
 ## Release notes
 
-( ) This is not user-visible and no release notes are required.
+( ) This is not user-visible or docs only and no release notes are required.
```
Comment (findepi) [inline .github/pull_request_template.md:15]:
```suggestion
- [ ] The change is not user-visible, or is docs only change, and no release notes are required.
```

?

Comment (mosabua) [reply .github/pull_request_template.md:15]: or affects documentation only

**Comment by colebow** [comment]
So as far as the checkboxes are concerned:

- I think the upgrade they provide is minimal (instead of typing x, you click)
- I think the downside they provide (surfacing an entirely new element of the GitHub UI, which is then misleading - see my first comment on this PR) is suboptimal and worth avoiding unless there is a very convincing reason

**Comment by findepi** [comment]
whether "(instead of typing x, you click" is "minimal" or "surfacing an entirely new element of the GitHub UI" is important sounds subjective

To me, "instead of typing x, you click" is non-negligible thing. How are you making sure you're subjective truth is more right?

**Pull request merged**